File: app/src/main/java/com/chooloo/www/callmanager/ui/search/SearchMvpPresenter.java
package com.chooloo.www.callmanager.ui.search;

import com.chooloo.www.callmanager.ui.base.BasePresenter;
import com.chooloo.www.callmanager.ui.base.MvpPresenter;

public interface SearchMvpPresenter<V extends SearchMvpView> extends MvpPresenter<V> {
    void onTextChanged(String text);

    void onFocusChanged(boolean hasFocus);
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/search/SearchMvpPresenter.kt
package com.chooloo.www.callmanager.ui.search

import com.chooloo.www.callmanager.ui.base.MvpPresenter

interface SearchMvpPresenter<V : SearchMvpView> : MvpPresenter<V> {

    fun onTextChanged(text: String)
    fun onFocusChanged(hasFocus: Boolean)
}